src/main/java/com/github/sardine/impl/SardineImpl.java

```diff
@@ -237,8 +243,23 @@ private CredentialsProvider createDefaultCredentialsProvider(String username, St
 	@Override
 	public void enableCompression()
 	{
-		this.builder.addInterceptorLast(new RequestAcceptEncoding());
-		this.builder.addInterceptorLast(new ResponseContentEncoding());
+		// Adding content encoding (gzip) support
+		this.builder.addRequestInterceptorLast((httpRequest, entityDetails, context) -> {
+			httpRequest.addHeader("Accept-Encoding", "gzip");
+		});
+
+		this.builder.addResponseInterceptorLast((httpResponse, entityDetails, context) -> {
+			ClassicHttpResponse classicHttpResponse = (ClassicHttpResponse) httpResponse;
+			HttpEntity entity = classicHttpResponse.getEntity();
+
+			if (httpResponse.containsHeader("Content-Encoding") &&
+					"gzip".equalsIgnoreCase(httpResponse.getHeader("Content-Encoding").getValue())) {
+				// Decompressing the gzip content
+				ClassicHttpResponse.class.cast(httpResponse).setEntity(
+						new GzipDecompressingEntity(entity));
+			}
+		});
+
```

Comment:
Is there no longer a "simple" way to do this?

Reply:
Should just toggle `contentCompressionDisabled` in the client builder instead.

Reply:
There is no setter in the builder to enable. Can be just a no-op then as it is enabled by default?